tests/test_endpoint_buffers.py

```diff
@@ -38,7 +39,172 @@ def forward_setup_fixture():
+            area=area,
+            should_export_plots=False)
+
+        output = endpoint_buffer.prepare_results_for_publish()
+
+        assert output == {
+            "job_id": "JOB_1",
+            "current_market": "",
+            "current_market_ui_time_slot_str": "",
+            "random_seed": 41,
+            "status": "",
+            "progress_info": {
+                "eta_seconds": 0,
+                "elapsed_time_seconds": 0,
+                "percentage_completed": 0,
+            },
+            "bids_offers_trades": {},
+            "results_area_uuids": [],
+            "simulation_state": {"general": {}, "areas": {}},
+            "simulation_raw_data": {},
+            "configuration_tree": {
+                "name": "area-name",
+                "uuid": "AREA",
+                "parent_uuid": "",
+                "type": "Area",
+                "children": []
+            }
+        }
+
+    @patch("gsy_e.gsy_e_core.sim_results.endpoint_buffer.get_json_dict_memory_allocation_size")
+    def test_prepare_results_for_publish_output_too_big(
+            self, get_json_dict_memory_allocation_size_mock, general_setup, caplog):
+        """The preparation of results fails if the output is too big."""
+        area, _ = general_setup
+        endpoint_buffer = SimulationEndpointBuffer(
+            job_id="JOB_1",
+            random_seed=41,
+            area=area,
+            should_export_plots=False)
+
+        get_json_dict_memory_allocation_size_mock.return_value = 128000
+        with caplog.at_level(logging.WARNING):
+            output = endpoint_buffer.prepare_results_for_publish()
+            assert "Do not publish message bigger than 64 MB, current message size 128.0 MB." \
+                in caplog.text
+
+        assert output == {}
+
+    def test_generate_json_report(self, general_setup):
+        area, _ = general_setup
+        endpoint_buffer = SimulationEndpointBuffer(
+            job_id="JOB_1",
+            random_seed=41,
+            area=area,
+            should_export_plots=False)
+
+        # The ResultsHandler class should be tested as a different unit, so we just mock its output
+        results_handler_mock = MagicMock(all_raw_results={"mocked-results": "some-results"})
+        endpoint_buffer.results_handler = results_handler_mock
+
+        assert endpoint_buffer.generate_json_report() == {
+            "job_id": "JOB_1",
+            "random_seed": 41,
+            "status": "",
+            "progress_info": {
+                "eta_seconds": 0,
+                "elapsed_time_seconds": 0,
+                "percentage_completed": 0
+            },
+            "simulation_state": {"general": {}, "areas": {}},
+            "mocked-results": "some-results"
+        }
+
+    def test_general_results_are_generated(  # pylint: disable-msg=too-many-locals
+            self, general_setup):
+        """Test results are being correctly generated with respect to
+        spot market"""
```

Comment:
I do not understand which unit you are really testing here. Can you please explain?

Reply:
Actually some tests are already written in correspondence to Forward Market. Most of the testing units are common for both Forward and normal simulation. I understood that there will be a corresponding unit for normal results being generated. I think I might got it wrong.
But I will have a detailed look once again,  otherwise I will remove it and add the test for update stats of normal simulation.